builder/vsphere/supervisor/step_import_image.go

```diff
@@ -0,0 +1,379 @@
+	// Defaults to `false`.
+	CleanImportedImage bool `mapstructure:"clean_imported_image"`
+}
+
+func (c *ImportImageConfig) Prepare() []error {
+	if c.ImportSourceURL == "" {
+		return nil
+	}
+
+	var errs []error
+	if c.ImportTargetLocationName == "" {
+		errs = append(errs, fmt.Errorf("config import_target_location_name is required for importing image"))
+	}
+
+	if strings.HasPrefix(c.ImportSourceURL, "https://") && c.ImportSourceSSLCertificate == "" {
+		errs = append(errs, fmt.Errorf("config import_source_ssl_certificate is required for https based source urls"))
+	}
+
+	switch c.ImportTargetImageType {
+	case "":
+		c.ImportTargetImageType = "ovf"
+	case "ovf":
+		// If it's already "ovf", do nothing.
+	default:
+		errs = append(errs, fmt.Errorf("unsupported ImportTargetImageType: %s", c.ImportTargetImageType))
+	}
+
+	if c.WatchImportTimeoutSec == 0 {
+		c.WatchImportTimeoutSec = DefaultWatchImageImportTimeoutSec
+	}
+	if c.ImportRequestName == "" {
+		c.ImportRequestName = ImportRequestDefaultNamePrefix + uuid.NewString()[:5]
+	}
+
+	return errs
+}
+
+type StepImportImage struct {
+	ImportImageConfig *ImportImageConfig
+
+	ImportItemResourceName, Namespace string
+	TargetItemType                    imgregv1.ContentLibraryItemType
+	KubeWatchClient                   client.WithWatch
+}
+
+func (s *StepImportImage) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	logger := state.Get("logger").(*PackerLogger)
+
+	var err error
+	defer func() {
+		if err != nil {
+			state.Put("error", err)
+		}
+	}()
+
+	if err = s.initStep(state, logger); err != nil {
+		logger.Error("failed to initialize image import: %s", err.Error())
+		return multistep.ActionHalt
+	}
+
+	if err = s.validate(ctx, logger); err != nil {
+		logger.Error("failed to validate import image configs: %s", err.Error())
+		return multistep.ActionHalt
+	}
+
+	logger.Info("Importing the source image from %s to %s.",
+		s.ImportImageConfig.ImportSourceURL, s.ImportImageConfig.ImportTargetLocationName)
+
+	if err = s.createImageImportRequest(ctx, logger); err != nil {
+		return multistep.ActionHalt
+	}
+	state.Put(StateKeyImageImportRequestCreated, true)
+
+	if err = s.watchItemImport(ctx, state, logger); err != nil {
+		return multistep.ActionHalt
+	}
+
+	logger.Info("Finished importing the image from %s to %s.",
+		s.ImportImageConfig.ImportSourceURL, s.ImportImageConfig.ImportTargetLocationName)
+
+	return multistep.ActionContinue
+}
+
+func (s *StepImportImage) validate(ctx context.Context, logger *PackerLogger) error {
+	logger.Info("Validating image import request...")
+
+	var err error
+	if err = s.checkImportFeatureEnabled(ctx, logger); err != nil {
+		return err
+	}
+
+	if err = s.checkImportTarget(ctx, logger); err != nil {
+		return err
+	}
+
+	logger.Info("Image import configs are valid.")
+	return nil
+}
+
+func (s *StepImportImage) Cleanup(state multistep.StateBag) {
+	if state.Get(StateKeyImageImportRequestCreated) == false {
```

Comment:
hmm... I believe this will always be false as interface{} does not equal the boolean false. We need to use GetOk here. 
```suggestion
	if v, ok := state.GetOk(StateKeyImageImportRequestCreated); !ok || v == false {
```

Reply:
> hmm... I believe this will always be false as interface{} does not equal the boolean false. We need to use GetOk here.

I'll even suggest doing `!ok || !v` :) That being said, since `v` is an `interface{}` (unless if casted into a bool), doesn't that have the same problem?

Reply:
Ah yes - you need to type assert. I updated.

Reply:
Good catch, I'll update as suggested.